smooth/framework/functions/update_annuities.py

```diff
@@ -1,31 +1,28 @@
-
-
 def update_annuities(component):
     # Convert the CAPEX and variable costs to annuities.
     # Parameter:
     #  component: object of one component.
 
-    # First calculate the annuities for the CAPEX.
-    capex = component.capex
-    # When the capex dict is empty, the annuity is zero, otherwise it has to be calculated.
-    if not capex:
-        # There are no CAPEX, so the annuity is 0 [EUR/a].
-        capex_annuity = 0
-    else:
-        # Interest rate [-].
-        interest_rate = component.sim_params.interest_rate
-        # Calculate the capital recovery factor [-].
-        capital_recovery_factor = (interest_rate * (1 + interest_rate)**component.life_time) / \
-                                  (((1 + interest_rate)**component.life_time) - 1)
-        # Calculate the annuity of the CAPEX [EUR/a].
-        capex_annuity = capex['cost'] * capital_recovery_factor
-
+    # First calculate the annuities for the CAPEX in EUR/a.
+    # If there are no CAPEX (dict is empty), the annuity is 0 EUR/a,
+    # otherwise it is a product of capex and capital recovery factor [-].
+    capex_annuity = calc_annuity(component, component.capex)
     # Check if OPEX were calculated, if so they are directly in annuity format.
     if not component.opex:
         opex = 0
     else:
         opex = component.opex['cost']
 
+    # Calculate the emission annuity for the installation in kg/a.
+    # If the emissions are not given (dict is empty), the annuity is 0 kg/a,
+    # otherwise it is a product of fix_emissions and capital recovery factor [-].
+    fix_emissions_annuity = calc_annuity(component, component.fix_emissions)

```

Comment:
There shouldn't be an interest rate for the CO2-Emissions, only for the costs. So there should be another function next to calc_annuity for calcluation the emissions without the interest rate.

Reply:
ok

Reply:
https://github.com/rl-institut/smooth/pull/15/commits/3760d04c1db9859f0166b6280cb07c26bbb926ed